sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -923,6 +925,19 @@ class Analyzer(
 
       case u: UnresolvedV2Relation =>
         CatalogV2Util.loadRelation(u.catalog, u.tableName).getOrElse(u)
+
+      case u @ UnresolvedRelation(_, extraOptions, true) =>
+        val r = expandRelationName(u.multipartIdentifier) match {
+          case NonSessionCatalogAndIdentifier(catalog, ident) =>
+            CatalogV2Util.loadTable(catalog, ident) match {
+              case Some(table) =>
+                Some(StreamingRelationV2(
```

Comment:
With the refactory of StreamingRelationV2(#29633), we can directly create it in the catalyst.

Reply:
shall we just add a `isStreaming` flag in `lookupV2Relation`, to unify the code a bit more?

Reply:
Sure, done in 305c316